### PR TITLE
Changes required for remote action support 

### PR DESCRIPTION
**Pull request by ettec**
Changes required for remote action support  as part of -> https://smartcontract-it.atlassian.net/browse/CAPPL-108

### CODE REVIEW

File: pkg/capabilities/capabilities.go [hunk 1]
```diff
@@ -71,6 +71,7 @@ type RequestMetadata struct {
 type RegistrationMetadata struct {
 	WorkflowID    string
 	WorkflowOwner string
+	ReferenceID   string
```
Comment (krehermann) [inline pkg/capabilities/capabilities.go:74]: reference to what? inline comment?

File: pkg/capabilities/pb/capabilities_helpers.go [hunk 2]
```diff
@@ -149,6 +149,106 @@ func UnmarshalTriggerResponse(raw []byte) (capabilities.TriggerResponse, error)
...
+	}
+
+	config, err := values.FromMapValueProto(req.Config)
+	if err != nil {
+		return capabilities.RegisterToWorkflowRequest{}, err
+	}
+
+	return capabilities.RegisterToWorkflowRequest{
+		Metadata: capabilities.RegistrationMetadata{
+			WorkflowID:  req.Metadata.WorkflowId,
```
Comment (krehermann) [inline pkg/capabilities/pb/capabilities_helpers.go:183]: why isn't owner populated in any of these calls?

File: pkg/capabilities/capabilities.go [hunk 3]
```diff
@@ -369,7 +374,8 @@ func (c *RemoteTriggerConfig) ApplyDefaults() {
 }
 
 type CapabilityConfiguration struct {
-	DefaultConfig       *values.Map
-	RemoteTriggerConfig *RemoteTriggerConfig
-	RemoteTargetConfig  *RemoteTargetConfig
+	DefaultConfig          *values.Map
+	RemoteTriggerConfig    *RemoteTriggerConfig
+	RemoteTargetConfig     *RemoteTargetConfig
+	RemoteExecutableConfig *RemoteExecutableConfig
```
Comment (cedric-cordenier) [inline pkg/capabilities/capabilities.go:380]: Can we be consistent with the naming here -> RemoteExecutableConfig -> RemoteActionConfig?

File: pkg/capabilities/pb/capabilities.proto [hunk 4]
```diff
@@ -80,6 +80,7 @@ message CapabilityResponse {
 
 message RegistrationMetadata {
   string workflow_id = 1;
```
Comment (cedric-cordenier) [inline pkg/capabilities/pb/capabilities.proto:82]: + Owner is missing; can you add?

**Review by krehermann — APPROVED** [review]

File: pkg/capabilities/pb/capabilities_helpers.go [hunk 5]
```diff
@@ -149,6 +149,106 @@ func UnmarshalTriggerResponse(raw []byte) (capabilities.TriggerResponse, error)
...
+func RegisterToWorkflowRequestToProto(req capabilities.RegisterToWorkflowRequest) *RegisterToWorkflowRequest {
+	config := values.EmptyMap()
+	if req.Config != nil {
+		config = req.Config
+	}
+
+	return &RegisterToWorkflowRequest{
+		Metadata: &RegistrationMetadata{
+			WorkflowId:  req.Metadata.WorkflowID,
+			ReferenceId: req.Metadata.ReferenceID,
```
Comment (cedric-cordenier) [inline pkg/capabilities/pb/capabilities_helpers.go:161]: + Owner

File: pkg/capabilities/capabilities.go (hunk 1 above)
Comment (ettec) [reply pkg/capabilities/capabilities.go:74]: good point, I was tempted to update to StepReferenceID, but ReferenceID is already used elsewhere  for the same so went for consistency

File: pkg/capabilities/pb/capabilities_helpers.go (hunk 2 above)
Comment (ettec) [reply pkg/capabilities/pb/capabilities_helpers.go:183]: honestly, didn't know if it was needed (whereas ReferenceID def is), but I detect a groundswell of support in the comment for adding it ;-), will add it

File: pkg/capabilities/capabilities.go (hunk 1 above)
Comment (ettec) [reply pkg/capabilities/capabilities.go:74]: comment/rename?  I'll go comment unless objections

File: pkg/capabilities/capabilities.go (hunk 3 above)
Comment (ettec) [reply pkg/capabilities/capabilities.go:382]: @cedric-cordenier based on corresponding core review you're ok with leaving this as is?

**Review by ilija42 — APPROVED** [review]

**Review by cedric-cordenier — APPROVED** [review]

File: pkg/capabilities/capabilities.go (hunk 3 above)
Comment (cedric-cordenier) [reply pkg/capabilities/capabilities.go:382]: I am, but please file a follow up so we can revisit this later

**Review by george-dorin — APPROVED** [review]

**Pull request merged**